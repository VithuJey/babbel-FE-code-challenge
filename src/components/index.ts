import Layout from "./layout/Layout";
import Card from "./card/Card";
import Modal from "./modal/Modal";

export { Layout, Card, Modal };
