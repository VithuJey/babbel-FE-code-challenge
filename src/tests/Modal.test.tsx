import React, { useState } from "react";
import { create } from "react-test-renderer";
import { Modal } from "../components";

describe("Modal component", () => {
  test("Matches the snapshot", () => {
    const modal = create(
      <Modal
        showModal={true}
        setShowModal={() => {}}
        winner={{
          id: "12345",
          name: "Player 1",
          imageUrl: "../../public/1.png",
        }}
        getGameDetails={async () => {}}
      />
    );
    expect(modal.toJSON()).toMatchSnapshot();
  });

  test("Modal renders player name", () => {
    const playerName = "Player 1";
    const modal = create(
      <Modal
        showModal={true}
        setShowModal={() => {}}
        winner={{
          id: "12345",
          name: playerName,
          imageUrl: "../../public/1.png",
        }}
        getGameDetails={async () => {}}
      />
    );
    const modalInstance = modal.root;
    expect(modalInstance.findByProps({ className: "name" }).children).toEqual([
      playerName,
    ]);
  });

  test("Modal is opened", () => {
    const modal = create(
      <Modal
        showModal={true}
        setShowModal={() => {}}
        winner={{
          id: "12345",
          name: "Player 1",
          imageUrl: "../../public/1.png",
        }}
        getGameDetails={async () => {}}
      />
    );

    const modalInstance = modal.root;
    const childElement = modalInstance.findByType("div");
    expect(childElement.props.className).toEqual("modal-block");
  });

  test("Modal is closed", () => {
    const modal = create(
      <Modal
        showModal={false}
        setShowModal={() => {}}
        winner={{
          id: "12345",
          name: "Player 1",
          imageUrl: "../../public/1.png",
        }}
        getGameDetails={async () => {}}
      />
    );

    const modalInstance = modal.root;
    const childElement = modalInstance.findByType("div");
    expect(childElement.props.className).toEqual("modal-none");
  });
});
