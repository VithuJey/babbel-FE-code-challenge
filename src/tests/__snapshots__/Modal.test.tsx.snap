// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Matches the snapshot 1`] = `
<div
  className="modal-block"
>
  <div
    className="modal-content"
  >
    <span
      className="close"
      onClick={[Function]}
    >
      Ã—
    </span>
    <h2>
      We got a winner!
    </h2>
    <img
      alt="avatar"
      src="../../public/1.png"
    />
    <p
      className="name"
    >
      Player 1
    </p>
    <button
      onClick={[Function]}
    >
      Roll again
    </button>
  </div>
</div>
`;
